src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchSource.java

```diff
@@ -122,36 +125,48 @@ public void prepareRun(BatchSourceContext context) {
     } catch (ConnectionException exception) {
       String message = SalesforceConnectionUtil.getSalesforceErrorMessageFromException(exception);
       collector.addFailure(String.format("Unable to get organization Id due to error: %s", message),
-                           "Ensure Credentials are correct.");
+          "Ensure Credentials are correct.");
     }
     Asset asset = Asset.builder(config.getReferenceNameOrNormalizedFQN(orgId, sObjectName))
-      .setFqn(config.getFQN(orgId, sObjectName)).build();
+        .setFqn(config.getFQN(orgId, sObjectName)).build();
     LineageRecorder lineageRecorder = new LineageRecorder(context, asset);
     lineageRecorder.createExternalDataset(schema);
     lineageRecorder.recordRead("Read", "Read from Salesforce",
-                               Preconditions.checkNotNull(schema.getFields()).stream()
-                                 .map(Schema.Field::getName)
-                                 .collect(Collectors.toList()));
+        Preconditions.checkNotNull(schema.getFields()).stream()
+            .map(Schema.Field::getName)
+            .collect(Collectors.toList()));
 
     authenticatorCredentials = config.getConnection().getAuthenticatorCredentials();
+    List<SalesforceSplit> querySplits =
+        getSplits(config, authenticatorCredentials, context.getLogicalStartTime(), oAuthInfo);
+    querySplits.parallelStream().forEach(salesforceSplit -> jobIds.add(salesforceSplit.getJobId()));
```

Comment:
jobIds is not thread-safe, so please don't use parallelism. Why do you need parallelism here?